torch/library.h

```diff
@@ -605,12 +605,18 @@ class TORCH_API Library final {
   /// }
   /// ```
 
-  template <typename Schema>
   Library& def(
-      Schema&& raw_schema,
+      c10::FunctionSchema s,
```

Comment:
Why not pass it by reference if we expect it to be moved anyway?

Reply:
pass-by-value this is a succinct way to implement "move if we can, copy if we have to". however, I checked the C++ core guidelines, and it appears they just [recommend](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f15-prefer-simple-and-conventional-ways-of-passing-information) the more efficient approach of having two overloads, one for const ref and one for rvalue ref, so I'll do that.

Reply:
turns out this wouldn't have built in the first place with `const c10::FunctionSchema&`, so we can just use rvalue ref directly.